File: resources/matlab/bioinspirados/Eggholder.m
%------------------------------------------%
% Imput                                    %
% X = [x1, x2]                    %
%------------------------------------------%
function Y = Eggholder (X) % 
    % Eggholder function 
    x1 = X(1);
    x2 = X(2);

   part1 = -(x2+47) * sin(sqrt(abs(x2+x1/2+47)));
   part2 = -x1 * sin(sqrt(abs(x1-(x2+47))));

   Y = part1 + part2;
end

File: ene300/functions/_eggholder.py
import numpy as np
from ._function_counter import function_counter

@function_counter
def eggholder(x):
    # Reference
    # Global Optimization Test Problems. Retrieved June 2013, from http://www-optima.amp.i.kyoto-u.ac.jp/member/student/hedar/Hedar_files/TestGO.htm. 
    # EASOM FUNCTION Funções para testes de Algoritmos
    y = x[1]
    x = x[0]
    return -(y + 47) * np.sin(np.sqrt(np.abs(x/2 + y + 47))) - x * np.sin(np.sqrt(np.abs(x - (y + 47))))
